en/editor/editor-customization/engine-customization.md

```diff
@@ -0,0 +1,101 @@
+# Engine Customization Workflow
+
+The engine part of Cocos Creator 3.0 includes JavaScript, Cocos2d-x-lite, and adapter (adapter engine customization is not supported at this time). The engine is all open-source on GitHub. The addresses are as follows:
+ 
+- JavaScript engine: <https://github.com/cocos-creator/engine>
+- Cocos2d-x-lite engine: <https://github.com/cocos-creator/cocos2d-x-lite>
+- Adapter: <https://github.com/cocos-creator-packages/adapters>
+
+It is recommended that you maintain your custom code through GitHub's Fork workflow so that you can easily update the custom parts when the engine is upgraded in the future, as described in [Fork a repo](https://help.github.com/articles/fork-a-repo). If you'd like to help Cocos get better and better, feel free to submit your changes to GitHub, see [How to Submit Code to Cocos](../../submit-pr/submit-pr.md). For more GitHub-related workflows, please refer to [GitHub Help](https://help.github.com).
+
+Also, depending on the Creator version, you will need to switch to a different engine branch, for example:
+
+- **3d-vX.Y** branch: the branch that corresponds to the X.Y version
+
+It is usually recommended to use the same branch as the version of the Creator used.
```

Comment:
It is usually recommended to use the same branch as the version of the Creator used.
-->
It is recommended to use the same branch that corresponds to the version of Creator being used.